Footer

martink-rsa opened the pull request:
1. Added the footer according to AC.
2. Changed both header and footer to pixel perfect design using oolioo.io technique. Possibility it is not perfect as I wasn't using a reliable crop of the Zeplin design. Better Zeplin designs will be used once I have thought of a reliable and quick way to transfer from Zeplin to the overlay browser extension.
3. 1. Removed styling from Link component and added it to `a` global styles
    2. Changed Link component so that `<Link>` can be used interchangeably between `a` and Router `Link`s.

martink-rsa committed:
Added footer, pixel-perfect design

jkettmann commented on public/index.html:
```diff
@@ -24,16 +22,14 @@
-    <div id="root"></div>
-    <!--
+  <link href="https://fonts.googleapis.com/css?family=Bitter:400|Montserrat:400,300&display=swap" rel="stylesheet" />
+  <title>React App</title>
```
I should add it in a ticket but I guess it's good to rename this to "Reddit Timer"

jkettmann commented on src/components/App/App.js:
```diff
@@ -24,9 +27,22 @@ function App() {
       },
     ],
   };
+  const footerObj = {
```
I think it might be better to move this into the footer. The same goes for the header as well.

To me, this looks a bit like unnecessary abstraction. We won't change the logo or links inside the header and footer. So we don't need to pass them down as props.

Another abstraction that's not necessary (and in fact causes more harm than good in the footer) is the links array. Honestly, I would just hard-code them inside the footer (and probably inside the header as well). I'll write more about a possible solution for the different links (external and internal) in the footer and Link component

jkettmann commented on src/components/Footer/Footer.js:
```diff
@@ -0,0 +1,44 @@
+  return (
+    <FooterWrapper>
+      <FooterLeft>
+        <Link href={links[0].link} alt={links[0].title}>
```
Here it seems much easier to just hard-code the href and alt. Imagine reading this in the future. You can't just look at this file and understand where this link is pointing to and what it says. Now compare that to this

```
<Link href="https://ooloo.io" alt="ooloo.io">
  ooloo.io
</Link>
```

Now it's immediately clear. And since we won't use the footer with a different set of links this seems like an unnecessary layer of abstraction.

jkettmann commented on src/components/Footer/Footer.style.js:
```diff
@@ -0,0 +1,48 @@
+`;
+
+const FooterCenter = styled(FooterItem)`
+padding-left: 1px;
```
Missing indentation. Same one line below

jkettmann commented on src/components/Footer/Footer.style.js:
```diff
@@ -0,0 +1,48 @@
+  font-size: 14px;
+`;
+
+const FooterLeft = styled(FooterItem)`
```
Great way of sharing common styles

jkettmann commented on src/components/Header/Header.js:
```diff
@@ -1,23 +1,23 @@
       <LinksContainer>
         {links.map((item) => (
-          <Link key={item.title} to={item.link}>{item.title}</Link>
+          <NavLink key={item.title} to={item.link} alt={item.title}>{item.title}</NavLink>
```
Here it makes more sense to have an array of links. But I'd still move the `links` prop here. You can have it as a constant outside of Header component. This way it won't be re-initialized every time the component is rendered

```
const links = [...]

function Header({ logo, links }) { 
```

jkettmann commented on src/components/Link/Link.js:
```diff
@@ -1,16 +1,28 @@
+  to, href, alt, className, children,
+}) {
+  // If there is a 'href', return a normal a href tag, otherwise use Router
+  return href ? <a className={className} href={href} alt={alt}>{children}</a>
```
Instead of this conditional logic you can use the [as prop](https://styled-components.com/docs/api#as-polymorphic-prop) to use another component. So if you have your previous Link implementation

```
const Link = styled(link)`...`
``` 

you can have an external link inside the footer like this

```
<Link 
  as="a"
  href="https://ooloo.io" 
  alt="ooloo.io"
>
  ooloo.io
</Link>
```

Then you can also remove the nasty global styles :-)

jkettmann commented on src/styles/globalStyle.js:
```diff
@@ -1,14 +1,31 @@
+  h2 {
+    font-size: 1.5rem;
+  }
+  a {
```
See my comment in Link.js

jkettmann reviewed (changes requested):
Good job overall. I left a few comments.

Can you also change the title of the PR to "Footer". It's common to use the title of the ticket rather than a description of what you did. Then it's easier to understand what changes belong to which ticket later on when you need to understand the git history for some reason.

jkettmann commented:
I just realized that there is a `package-lock.json` file. Have you been using npm? Would be great if you could switch to yarn instead. I'll probably set up automatic PR deployment for all the repos and need consistent scripts for that. Otherwise, we end up using different lock file which can lead to nasty errors.

martink-rsa committed:
Footer

martink-rsa commented:
I have taken all the suggestions into account.

1. Renamed HTML title and gave site description.
2. Fixed indentation.
3. Removed abstraction for Header and Footer links with exception of the 3 nav links that get `map`ped.
4. Changed `Link` to only be a styled-component with no logic and am using `as="a"` to trigger an `a` link. I tried to get this going myself but I was looking in the wrong docs for the answer (looked at Router docs, not styled-components).
5. Changed to `yarn` and deleted `package.lock`

jkettmann commented on src/components/Header/Header.js:
```diff
@@ -1,37 +1,42 @@
+  },
+];
+
+function Header() {
```
I had a look in the browser now and saw that you can simplify the header quite a bit. Currently the header looks like this

```
<header>
  <div>
    <a alt="" href="/">
      <img src="/static/media/logo.1528c9cf.svg" alt="Logo">
    </a>
  </div>
  <nav>
    <a alt="Search" href="/search?title=&quot;javascript&quot;">Search</a>
    <a alt="How it works" href="/#how-it-works">How it works</a>
    <a alt="About" href="/#about">About</a>
  </nav>
</header>
```

You could simplify it to be like this

```
<header>
  <a alt="" href="/">
    <img src="/static/media/logo.1528c9cf.svg" alt="Logo">
  </a>

  <nav>
    <a alt="Search" href="/search?title=&quot;javascript&quot;">Search</a>
    <a alt="How it works" href="/#how-it-works">How it works</a>
    <a alt="About" href="/#about">About</a>
  </nav>
</header>
```

- <header> should have padding: 0 80px, display: flex, align-items: center, justify-content: space-between => would get rid of the LogoContainer
- <nav> doesn't need align-items, justify-content and padding
- the logo link should not be a NavLink since its styling needs to be different

This way you simplified the element a bit, but especially the CSS will benefit

jkettmann commented on src/components/Header/Header.style.js:
```diff
@@ -1,26 +1,32 @@
+`;
+
+const NavLink = styled(Link)`
+  margin-left: 25px;
```
The text color is not correct

jkettmann commented on src/components/Footer/Footer.style.js:
```diff
@@ -0,0 +1,48 @@
+  width: 100%;
+`;
+
+const FooterItem = styled.div`
```
The footer looks good but the text color is not correct

jkettmann commented on src/pages/HomePage/HomePage.style.js:
```diff
@@ -0,0 +1,11 @@
+display: flex;
+align-items: center;
+flex-direction: column;
+  height: 100%;
```
You can remove height. See my comment in globalStyle

jkettmann commented on src/components/Header/Header.style.js:
```diff
@@ -1,26 +1,32 @@
 const HeaderWrapper = styled.header`
   display: flex;
-  height: 65px;
+  min-height: 100px;
```
This should be height, not min-height.  See my comment in globalStyle

jkettmann commented on src/styles/globalStyle.js:
```diff
@@ -1,14 +1,23 @@
-    font-family: ${(props) => props.theme.fonts[0]};
-    color: ${(props) => props.theme.colors.textSecondary.main};
+  #root, html, body {
+    height: 100%;
```
If you want the app to be at least a full window height (which is great) you can set `min-height: 100vh` and remove height here. Should fix the height issues in `HomePage` and `Header` as well

jkettmann reviewed (changes requested):
I ran the app and found a couple of smaller issues. 

Regarding the screenshot from Zeplin: There should be a keyboard shortcut to create a screenshot of an area of the screen. I'm on Ubuntu where I press ctrl+shift+PrtSc. Here is a [link for Mac](https://support.apple.com/en-us/HT201361). Then you can paste the screenshot to the Perfect-Pixel from the clipboard

martink-rsa committed:
Footer

martink-rsa commented:
There will be some artefacts in this last commit related to the Hero. Please ignore them for now.

I've made the following changes:
1. Cleaned up header so it follows your suggested structure. This cleaned up a lot of the spacing I had added.
2. Changed the app size to 100vh, then set the content window (HomePage, Search etc.) to `flex-grow: 1`. I'm not sure I've done what you wanted to see and could have possibly done the wrong thing.
2. Changed the colours of the links as well as some other colours. I am still trying to figure out a good place in the theme to put the link colour and will adjust it in the near future once I've thought of a good solution.

martink-rsa committed:
Footer

martink-rsa committed:
Footer

martink-rsa committed:
Footer

martink-rsa committed:
Footer
